### ISSUE TITLE
[Bug]: failure to load High-Dimensional HNSW Index with large collection

### ISSUE THREAD

**Issue by douglarek** [milvus-io/milvus#31993]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.13
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory:  It is calculated using the mivlus tool sizing
- GPU: 
- Others:
```

### Current Behavior

Attempting to load a collection of approximately 2.6 million items, utilizing an HNSW (Hierarchical Navigable Small World) index with a dimensionality of 4096, is resulting in a failure when loading the collection into Milvus.

**Error Message:**
```
I20240407 13:40:40.517884 10927 VectorMemIndex.cpp:492] [SEGCORE][LoadFromFile][milvus] load index into Knowhere...
Assert "data_ != MAP_FAILED" at /go/src/github.com/milvus-io/milvus/internal/core/src/mmap/Column.h:85
 => failed to create file-backed map, err: Invalid argument
[2024/04/07 13:40:40.579 +00:00] [WARN] [segments/cgo_util.go:54] ["CStatus returns err"] [error="Assert \"data_ != MAP_FAILED\" at /go/src/github.com/milvus-io/milvus/internal/core/src/mmap/Column.h:85\n => failed to create file-backed map, err: Invalid argument"]
Assert "data_ != MAP_FAILED" at /go/src/github.com/milvus-io/milvus/internal/core/src/mmap/Column.h:85
 => failed to create file-backed map, err: Invalid argument
[2024/04/07 13:40:40.616 +00:00] [WARN] [segments/cgo_util.go:54] ["CStatus returns err"] [error="Assert \"data_ != MAP_FAILED\" at /go/src/github.com/milvus-io/milvus/internal/core/src/mmap/Column.h:85\n => failed to create file-backed map, err: Invalid argument"]
I20240407 13:40:40.620813 10995 SegmentSealedImpl.cpp:260] [SEGCORE][LoadFieldData][milvus] finish loading segment field, segmentID:448914920945483722, fieldID:104
[2024/04/07 13:40:40.620 +00:00] [INFO] [segments/segment.go:727] ["load field done"] [traceID=72fbb2dfbd26fb638202b3c211bc8fae] [collectionID=448914920947507608] [partitionID=448914920947507609] [segmentID=448914920945483722] [fieldID=104] [rowCount=64000]
[2024/04/07 13:40:40.624 +00:00] [INFO] [gc/gc_tuner.go:90] ["GC Tune done"] ["previous GOGC"=200] ["heapuse "=24] ["total memory"=2466] ["next GC"=55] ["new GOGC"=200] [gc-pause=167.355µs] [gc-pause-end=1712497240624340604]
[2024/04/07 13:40:40.627 +00:00] [WARN] [segments/segment_loader.go:265] ["load segment failed when load data into memory"] [traceID=72fbb2dfbd26fb638202b3c211bc8fae] [collectionID=448914920947507608] [segmentType=Sealed] [partitionID=448914920947507609] [segmentID=448914920945483722] [error="Assert \"data_ != MAP_FAILED\" at /go/src/github.com/milvus-io/milvus/internal/core/src/mmap/Column.h:85\n => failed to cre
```
### Expected Behavior

The collection should load successfully into Milvus without encountering any errors related to creating file-backed maps.

### Steps To Reproduce

```
1. Create a collection with approximately 2.6 million items.
2. Use an HNSW index with a dimensionality of 4096.
3. Attempt to load the collection into Milvus.
```

### Milvus Log

[querynode.log](https://github.com/milvus-io/milvus/files/14901284/querynode.log)

**Comment by yanliang567** [comment]
/assign @sunby 
is it something about mmap?

/unassign

**Comment by xiaofan-luan** [comment]
> failed to create file-backed map, err: Invalid argument

Did you set a correct local storage path for mmap?

**Comment by xiaofan-luan** [comment]
hard to tell why invalid argument, but my guess is due to wrong directory or access

**Comment by xiaofan-luan** [comment]
i think that's probably the problem. You need to set a valid path

**Comment by xiaofan-luan** [comment]
collection level config simply enable/disable mmap

**Comment by sunby** [comment]
We've addressed a similar issue before, where if a user's column type is a string or array and all values in that column are empty, an error would occur during loading. We've fixed this issue in version 2.4.

BTW, if you want to disable mmap in 2.3.13, you should set `queryNode.mmapDirPath` to empty in server side.

**Comment by sunby** [comment]
And we have added an mmap switch `queryNode.mmapEnabled` in version 2.4. You can set it to false to disable mmap.

**Comment by yanliang567** [comment]
> > We've fixed this issue in version 2.4.
> 
> Will this fix be ported to version 2.3? For example, releasing version 2.3.14 with this fix included?

it only contains in 2.4 or later
